packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/blocs/onboarding_bloc.dart

```diff
@@ -0,0 +1,206 @@
+  /// Sets the currently entered email
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setEmail(String email);
+
+  /// Sets the currently entered password
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setPassword(String password);
+
+  /// Starts onboarding the user with the currently entered email and password
+  void register();
+
+  /// Resumes the onboarding process if the user already has an auth token
+  void resumeOnboarding();
+
+  void navigateToNextStep(UserWithAuthTokenModel user);
+}
+
+/// A contract class containing all states of the OnboardingBloC.
+abstract class OnboardingBlocStates {
+  /// The currently entered email state
+  Stream<String> get email;
+
+  /// The currently entered password state
+  Stream<String> get password;
+
+  /// State indicating whether the user is logged in successfully
+  ConnectableStream<UserWithAuthTokenModel> get registered;
+
+  /// The state indicating whether we show field errors to the user
+  Stream<bool> get showFieldErrors;
+
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  /// The error state for resuming the onboarding process
+  Stream<ErrorModel> get resumeOnboardingErrors;
+
+  /// The routing state
+  ConnectableStream<void> get onRouting;
+}
+
+@RxBloc()
+class OnboardingBloc extends $OnboardingBloc {
+  OnboardingBloc(
+    this._authService,
+    this._permissionsService,
+    this._onboardingService,
+    this._validatorService,
+    this._routerBloc,
+  ) {
+    registered.connect().addTo(_compositeSubscription);
+    onRouting.connect().addTo(_compositeSubscription);
+
+    resumeOnboardingErrors.listen((_) {});
+    _resumeOnboardingIfHasAuthToken();
+  }
+
+  Future<void> _resumeOnboardingIfHasAuthToken() async {
+    if (await _authService.isAuthenticated()) {
+      resumeOnboarding();
+    }
+  }
+
+  final AuthService _authService;
+  final PermissionsService _permissionsService;
+  final OnboardingService _onboardingService;
+  final LoginValidatorService _validatorService;
+  final RouterBlocType _routerBloc;
+
+  @override
+  Stream<String> _mapToEmailState() => _$setEmailEvent
+      .map(_validatorService.validateEmail)
+      .startWith('')
+      .shareReplay(maxSize: 1);
+
+  @override
+  Stream<String> _mapToPasswordState() => _$setPasswordEvent
+      .map(_validatorService.validatePassword)
+      .startWith('')
+      .shareReplay(maxSize: 1);
+
+  @override
+  ConnectableStream<UserWithAuthTokenModel> _mapToRegisteredState() => _$registerEvent
+      .throttleTime(const Duration(seconds: 1))
+      .withLatestFrom2<Result<String>, Result<String>, CredentialsModel?>(
+          email.asResultStream(),
+          password.asResultStream(),
+          (_, emailResult, passwordResult) => _validateAndReturnCredentials(
+                emailResult,
+                passwordResult,
+              ))
+      .where((args) => args != null)
```

Comment:
Please use whereType() to avoid the bang operator in the following functions: such as `args!.email`

Reply:
Done